imap/mboxlist.c

```diff
@@ -2156,6 +2156,12 @@ mboxlist_delayed_deletemailbox(const char *name, int isadmin,
     r = mboxlist_lookup_allow_all(name, &mbentry, NULL);
     if (r) goto done;
 
+    /* Don't allow users to delete non-IMAP mailboxes */
+    if (!isadmin && (mbtype_isa(mbentry->mbtype) != MBTYPE_EMAIL)) {
```

Comment:
Should the `MBOXLIST_DELETE_FORCE` flags allow to delete as non-admin?

Reply:
The flag is only allowed for admins, so we don't have to check for its presence here.